BlueprintUI/Sources/Layout/LayoutAttributes.swift

```diff
@@ -155,6 +163,7 @@ public struct LayoutAttributes {
         result.alpha = alpha * layoutAttributes.alpha
         result.isUserInteractionEnabled = layoutAttributes.isUserInteractionEnabled && isUserInteractionEnabled
         result.isHidden = layoutAttributes.isHidden || isHidden
+        result.tintAdjustmentMode = layoutAttributes.tintAdjustmentMode ?? tintAdjustmentMode
```

Comment:
The cascade behavior can be tricky for these kinds of attributes. Consider this scenario:

```swift
element
  .tintAdjustmentMode(.automatic)
  .tintAdjustmentMode(.dimmed)
```

Would you expect the element's mode to be dimmed, or normal?

Reply:
Yeah this is what gives me pause. Lifting this into the `LayoutAttributes` may be dangerous in terms of the cascading affect up the superview chain. https://github.com/square/Blueprint/pull/467#discussion_r1330630096

Reply:
I think I would expect

```
element
  .tintAdjustmentMode(.automatic)
  .tintAdjustmentMode(.dimmed)
```

to have the mode `.dimmed`.

But what about this case? Should the parent fallback to the child's mode if the parent's mode is `.automatic`? I'm not sure. The correct concatenation logic is kind of opaque to me.
```
element
  .tintAdjustmentMode(.normal)
  .tintAdjustmentMode(.automatic)
```

Is it appropriate for this element's mode to be `.normal`? 🤷

Reply:
> But what about this case? Should the parent fallback to the child's mode if the parent's mode is `.automatic`? I'm not sure. The correct concatenation logic is kind of opaque to me.
> 
> ```
> element
>   .tintAdjustmentMode(.normal)
>   .tintAdjustmentMode(.automatic)
> ```
> 
> Is it appropriate for this element's mode to be `.normal`? 🤷

I think so. I think we can kind of mimic the UIKit behavior here, so that the element inherits the nearest non-default value. So, when both `normal` and `dimmed` are present in the hierarchy, the nearest one wins. And any `automatic` has no real effect.

In that case the cascade would be something like:
```swift
switch tintAdjustmentMode {
  case .normal, .dimmed:
    result.tintAdjustmentMode = tintAdjustmentMode
  case .automatic:
    result.tintAdjustmentMode = layoutAttributes.tintAdjustmentMode
}
```

There's some precedent here in the other properties, I think. You can't reset opacity if a parent element reduces it, for example; you can only further reduce it.

Reply:
Given the precedence, I think this makes sense to me. Updated the concatenation logic and tests. Pushed in commit 4afb1cfc4d52a49e8e09ded64b08aea066f7247a.